packages/match-media/src/useBreakpointIndex.js

```diff
@@ -0,0 +1,35 @@
+import { useState, useEffect, useCallback } from 'react'
+import { useThemeUI } from 'theme-ui'
+
+const useBreakpointIndex = () => {
+  const {
+    theme: { breakpoints },
+  } = useThemeUI()
+
+  const getIndex = useCallback(() => {
+    let index = breakpoints.findIndex(
+      breakpoint => window.matchMedia(`(max-width: ${breakpoint})`).matches
+    )
+    if (index === -1) {
+      index = breakpoints.length - 1
+    }
+    return index
+  }, [breakpoints])
+
+  const [value, setValue] = useState(getIndex)
+
+  useEffect(() => {
+    const onResize = () => {
+      const newValue = getIndex()
+      if (value !== newValue) {
+        setValue(newValue)
+      }
+    }
+    window.addEventListener('resize', onResize)
```

Comment:
What were the issues with the match media listeners? I'm fairly sure that's a more performant way to handle this

Reply:
It was actually my first approach but I found it quite tricky and edge-casey, so for the sake of time constraints on a project that required it, I decided to instead opt for this simpler implementation, and in reality, it should still be very performant. It will only update state if the value actually changes, and I don't think calculating the breakpoints with `window.matchMedia` is a particularly CPU intensive process. It definitely seems more correct to use the listeners, but then again, for this particular use of them their API doesn't seem to match up very nicely.

Reply:
Yeah, no worries, just more curious what the edge cases might've been etc.